contracts/modules/capital/Pool.sol

```diff
@@ -275,8 +315,16 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
     swapDetails[assetAddress].lastSwapTime = lastSwapTime;
   }
 
-  function setSwapValue(uint newValue) external onlySwapOperator whenNotPaused {
-    swapValue = newValue.toUint96();
+  function setSwapAssetAmount(address assetAddress, uint value) external onlySwapOperator whenNotPaused {
+
+    assetsInSwapOperator[assetAddress] = value;
+
+    if (value > 0) {
+      uint assetId = getAssetId(assetAddress);
+      assetsInSwapOperatorBitmap = uint32(1 << assetId);
```

Comment:
Currently this isn't an issue because swap operator doesn't allow more than one order at a time, but I would insert a require statement checking that `assetsInSwapOperatorBitmap` is zero before overwriting it.